distributed/tests/test_worker.py

```diff
@@ -3502,3 +3505,43 @@ async def test_reconnect_argument_deprecated(s):
         warnings.simplefilter("error")
         async with Worker(s.address):
             pass
+
+
+@gen_test()
+async def test_worker_makes_own_thread():
+    class InitWorkerNewThread(WorkerPlugin):
+        name: name
```

Comment:
```suggestion
        name: str
```

not sure why mypy isn't allowing this, maybe because you're not in a typed context?

`async def test_worker_makes_own_thread() -> None: